File: Utils/ClassesGestioCitesMediques/src/org/milaifontanals/classes/MetgeEspecialitat.java
package org.milaifontanals.classes;

import java.io.Serializable;
import java.util.Objects;

public class MetgeEspecialitat implements Serializable{
    private int codiMetge;
    private int codiEspecialitat;
    
    private Metge codiEmpleat;
    private Especialitat codi;

    
    protected MetgeEspecialitat(){}

    public MetgeEspecialitat(Metge codiEmpleat, Especialitat codi) {
        setCodiEmpleat(codiEmpleat);
        setCodi(codi);
        
        codiEmpleat.addEspecialitat(this);
        codi.addMetge(this);
    }

    public Metge getCodiEmpleat() {
        return codiEmpleat;
    }

    public void setCodiEmpleat(Metge codiEmpleat) {
        this.codiEmpleat = codiEmpleat;
        codiMetge = codiEmpleat.getCodiEmpleat();
    }

    public Especialitat getCodi() {
        return codi;
    }

    public void setCodi(Especialitat codi) {
        this.codi = codi;
        codiEspecialitat = codi.getCodi();
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 67 * hash + Objects.hashCode(this.codiEmpleat);
        hash = 67 * hash + Objects.hashCode(this.codi);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final MetgeEspecialitat other = (MetgeEspecialitat) obj;
        if (!Objects.equals(this.codiEmpleat, other.codiEmpleat)) {
            return false;
        }
        if (!Objects.equals(this.codi, other.codi)) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "MetgeEspecialitat{" + "codiEmpleat=" + codiEmpleat + ", codi=" + codi + '}';
    }
    
    
}

File: GestioInformesHorari/GestioInformesHorariClasses/MetgeEspecialitat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GestioCitesMediquesClasses
{
    class MetgeEspecialitat
    {
        private int codiMetge;
        private int codiEspecialitat;

        private Metge codiEmpleat;
        private Especialitat codi;


        public MetgeEspecialitat(int codiMetge, int codiEspecialitat)
        {
            CodiMetge = codiMetge;
            CodiEspecialitat = codiEspecialitat;
        }

        public int CodiMetge { get => codiMetge; set => codiMetge = value; }
        public int CodiEspecialitat { get => codiEspecialitat; set => codiEspecialitat = value; }

        public override bool Equals(object obj)
        {
            return obj is MetgeEspecialitat me &&
                   codiMetge == me.codiMetge && 
                   codiEspecialitat == me.codiEspecialitat;
        }

        public override int GetHashCode()
        {
            return 1877310944 + codiMetge.GetHashCode() + codiEspecialitat.GetHashCode();
        }

        public override string ToString()
        {
            return codiMetge.ToString() + " " + codiEspecialitat.ToString();
        }
    }
}
